collect.py

```diff
@@ -119,25 +150,24 @@ def do_something(driver, elem_attributes=None):
             if not child.is_displayed() or not child.is_enabled():
                 continue
 
-            elem = child
-            break
+            res = check_attributes(driver, elem_attributes, child)  # returns list of elements with attributes = child_attributes
+            if len(res) == 1:
+                elem = child
+                break
+            else:
+                # remove all elements with the same attributes
+                print('Found elements with same attributes')
+                # print(res)
+                print()
+                for element in res:
+                    children.remove(element)
```

Comment:
Don't remove anything from the list, but define a list of elements to ignore and put them there.